Source/Core/VideoCommon/XFStructs.cpp

```diff
@@ -300,3 +301,431 @@ void PreprocessIndexedXF(u32 val, int refarray)
+                     i.embosssourceshift, i.embosslightshift);
+  }
+};
+
+template <>
+struct fmt::formatter<PostMtxInfo>
+{
+  constexpr auto parse(format_parse_context& ctx) { return ctx.begin(); }
+  template <typename FormatContext>
+  auto format(const PostMtxInfo& i, FormatContext& ctx)
+  {
+    return format_to(ctx.out(), "Index: {}\nNormalize before send operation: {}", i.index,
+                     i.normalize ? "Yes" : "No");
+  }
+};
+
+// These should maybe be moved to CPMemory.h
+template <>
+struct fmt::formatter<TMatrixIndexA>
+{
+  constexpr auto parse(format_parse_context& ctx) { return ctx.begin(); }
+  template <typename FormatContext>
+  auto format(const TMatrixIndexA& m, FormatContext& ctx)
+  {
+    return format_to(ctx.out(), "PosNormal: {}\nTex0: {}\nTex1: {}\nTex2: {}\nTex3: {}",
+                     m.PosNormalMtxIdx, m.Tex0MtxIdx, m.Tex1MtxIdx, m.Tex2MtxIdx, m.Tex3MtxIdx);
+  }
+};
+
+template <>
+struct fmt::formatter<TMatrixIndexB>
+{
+  constexpr auto parse(format_parse_context& ctx) { return ctx.begin(); }
+  template <typename FormatContext>
+  auto format(const TMatrixIndexB& m, FormatContext& ctx)
+  {
+    return format_to(ctx.out(), "Tex4: {}\nTex5: {}\nTex6: {}\nTex7: {}", m.Tex4MtxIdx,
+                     m.Tex5MtxIdx, m.Tex6MtxIdx, m.Tex7MtxIdx);
+  }
+};
+
+std::pair<std::string, std::string> GetXFRegInfo(u32 address, u32 value)
+{
+// Macro to set the register name and make sure it was written correctly via compile time assertion
+#define RegName(reg) ((void)(reg), #reg)
+#define DescriptionlessReg(reg) std::make_pair(RegName(reg), "");
+
+  switch (address)
+  {
+  case XFMEM_ERROR:
+    return DescriptionlessReg(XFMEM_ERROR);
+  case XFMEM_DIAG:
+    return DescriptionlessReg(XFMEM_DIAG);
+  case XFMEM_STATE0:  // internal state 0
+    return std::make_pair(RegName(XFMEM_STATE0), "internal state 0");
+  case XFMEM_STATE1:  // internal state 1
+    return std::make_pair(RegName(XFMEM_STATE1), "internal state 1");
+  case XFMEM_CLOCK:
+    return DescriptionlessReg(XFMEM_CLOCK);
+  case XFMEM_SETGPMETRIC:
+    return DescriptionlessReg(XFMEM_SETGPMETRIC);
+
+  case XFMEM_CLIPDISABLE:
+    return std::make_pair(
+        RegName(XFMEM_CLIPDISABLE),
+        fmt::format("{} clipping detection, {} trivial rejection, {} cpoly clipping acceleration",
+                    (value & 1) ? "Enable" : "Disable", (value & 2) ? "enable" : "disable",
+                    (value & 3) ? "enable" : "disable"));
+
+  case XFMEM_VTXSPECS:  //__GXXfVtxSpecs, wrote 0004
+    return std::make_pair(RegName(XFMEM_VTXSPECS),
+                          fmt::format("Vertex specs: {}", INVTXSPEC{.hex = value}));
+
+  case XFMEM_SETNUMCHAN:
+    return std::make_pair(RegName(XFMEM_SETNUMCHAN),
+                          fmt::format("Number of color channels: {}", value & 3));
+    break;
+
+  case XFMEM_SETCHAN0_AMBCOLOR:  // Channel Ambient Color
```

Comment:
Too bad we can't somehow pass the color from  here to Qt.  It'd be nice to somehow see the color in addition to the hex code

Reply:
I guess we could use rich text tags like I did for the tooltips.  Kinda muddies the description text though..